src/documentation/malloy_sql/index.md

```diff
@@ -0,0 +1,94 @@
+# MalloySQL
+
+MalloySQL enables mixing dialect-specific DDL with Malloy. For example, MalloySQL can be used to create a new table in BigQuery, based on a model of existing data:
+
+```malloy
+>>>malloy
+import "airports.malloy"
+
+>>>sql connection:bigquery
+CREATE TABLE major_airports AS ( // this is SQL
+  SELECT name
+  FROM %{
+    airports -> major_airports  // this is a Malloy query
+  }%
+);
+```
+
+## Usage
+
+MalloySQL can be used in the VSCode Extension and the Malloy CLI. In both cases, the extension of the file should be `.malloysql`.
+
+MalloySQL files are composed of two kinds of statements - Malloy statements and SQL statements. Each statement is preceeded by a line that defines the statement type: `>>>malloy` or `>>>sql`. The SQL statement can also be passed a connection name - this is the name of the connection to use when executing the query. This is valid MalloySQL:
+
+```malloy
+>>>sql connection:duckdb
+SELECT 1;
+```
+
+SQL statements can also contain embedded Malloy queries by wrapping the Malloy statement with `%{` and `}%`. To use embedded Malloy, a source must first be imported to use with the Malloy query:
+
+```malloy
+>>>malloy
+import "mysource.malloy"
+
+>>>sql connection:duckdb
+SELECT name_count from %{ mysource -> names }%;
+```
+
+It is possible to use multiple SQL commands in one statement (although only the final result will be shown in the VSCode extension):
+
+```malloy
+>>>malloy
+import "mysource.malloy"
+
+>>>sql connection:duckdb
+SELECT name_count from %{ mysource -> names }%;
+SELECT airport_count from %{ mysource -> airports }%;
+```
+
+The first SQL statement must define a connection to be used for queries, but future SQL statements only need to define a connection if a different connection should be used. This is valid:
+
+```malloy
+>>>malloy
+import "mysource.malloy"
+
+>>>sql connection:duckdb
+SELECT name_count from %{ mysource -> names }%;
+>>>sql
+SELECT airport_count from %{ mysource -> airports }%;
+```
+
+MalloySQL can contain multi (`/*...*/`) and single-line (`//` or `--`) comments.
+
+Running a specific SQL statement in a Malloy file (by, for example, clicking the "Run" codelens in the VSCode extension) will execute all preceeding Malloy statements, but only the selected SQL statement.
+
+## Example
+
+Here is a more realistic example, using a Malloy model to wrangle raw data and create a useful parquet file with DuckDB:
+
+```malloy
+>>>malloy
+import "raw_data.malloy"
+
+>>>sql connection:duckdb
+
+copy %{
+  raw_titles -> {
+    join_one: raw_ratings on tconst = raw_ratings.tconst
+    join_one: raw_crew on tconst = raw_crew.tconst
```

Comment:
I think in the next iteration of the language syntax, `join_one` is not going to be allowed in a query block, and will need to be added in an `extend` block (see the [examples in this doc](https://docs.google.com/document/d/1hMT8WIHvFjcxCoAosrJHvixEj-yJ7Q3Is0CwRyXHPCw/edit)).

It *is* legal syntax today, and I think it won't break even when the new syntax is implemented (both old and new syntax will work, but the old will throw warnings), but I would consider getting rid of these joins so there's one fewer place we need to update the docs when the changes land.

Reply:
Don't have doc access, requested it. In the interest of getting this in before i'm OOO a bunch I'm gonna push this - don't trust myself to get it exactly right anyways (I stole this from Lloyd). Would love any other example you think might be useful!